dhcpv4/bsdp/client.go

```diff
@@ -0,0 +1,113 @@
+package bsdp
+
+import (
+	"fmt"
+	"net"
+	"syscall"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+// Client is a BSDP-specific client suitable for performing BSDP exchanges.
+type Client dhcpv4.Client
+
+// BSDPExchange runs a full BSDP exchange (Inform[list], Ack, Inform[select],
+// Ack). Returns a list of DHCPv4 structures representing the exchange.
+func (c *Client) BSDPExchange(ifname string, d *dhcpv4.DHCPv4) ([]dhcpv4.DHCPv4, error) {
+	conversation := make([]dhcpv4.DHCPv4, 1)
+	var err error
+
+	// INFORM[LIST]
+	if d == nil {
+		d, err = NewInformListForInterface(ifname, dhcpv4.ClientPort)
+	}
+	conversation[0] = *d
+
+	fd, err := syscall.Socket(syscall.AF_INET, syscall.SOCK_RAW, syscall.IPPROTO_RAW)
```

Comment:
I guess I will move this code to a function in a future commit since it's basically duplicated from `dhcpv4.Client`, but not now. Could you add a TODO here for me please?